src/clients/FlexbaseClient.Card.ts

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Card } from '../models/Card/Card';
+import { Address } from '../models/Address/Address';
+import { CardOptions } from '../models/Card/CardOptions';
+import { FlexbaseClientBase } from './FlexbaseClient.Base';
+import { FlexbaseResponse } from '../models/FlexbaseResponse';
+
+interface QueryParameters {
+    searchTerm?: string;
+    status?: string;
+}
+
+interface CardsResponse extends FlexbaseResponse{
+    cards: Card[];
+}
+
+interface CardResponse extends FlexbaseResponse{
+    card: Card;
+}
+
+interface IssueCardForm {
+    cardType: string,
+    shipTo?: Address,
+    service?: string,
+}
+
+interface UpdateCardForm {
+    expensesTypes: {
+        amount: number,
+        groups: any,
+    };
+    notifyUse: boolean;
+    shipTo?: Address,
+    creditLimit: number,
+    cardName?: string,
+}
+
+export class FlexbaseClientCard extends FlexbaseClientBase {
+    private params({ searchTerm, status }: CardOptions = {}) {
+        const params: any = {};
+
+        if (typeof searchTerm === 'string') {
+            params.searchTerm = searchTerm;
+          }
+      
+          if (typeof status === 'string') {
+            params.status = status;
+          }
+
+        return params;
+    }
+
+    async getCompanyCards({ searchTerm, status }: QueryParameters = {}): Promise<Card[] | null> {
+        try {
+          const params = this.params({ searchTerm, status });
+    
+          const response = await this.client.url('/card/company').query(params).get().json<CardsResponse>();
+
+           if (!response.success) {
+              return null;
+           }
+    
+          return response.cards;
+        } catch (error) {
+          this.logger.error('Unable to get company cards', error);
+          return null;
+        }
+    }
+
+    async getUserCard(userId: string): Promise<Card | null> {
```

Comment:
What about `getCardsByUser`? @kyle-b2 @alegoos

Reply:
This endpoint returns the data for a specific card, the `id` isn't the `userId`, it is the `cardId`. My mistake. Changing now

Reply:
Also, what about `getCardById`?

Reply:
if it's just getting a card I would do `getCard` byId is inferred